kmultiversx/src/kmultiversx/kdist/mx-semantics/vmhooks/manBufOps.md

```diff
@@ -22,29 +22,29 @@ module MANBUFOPS
 
     syntax BytesResult ::= getBuffer(Int)                                   [function, total]
  // ---------------------------------------------------------------------------------------------
-    rule [[ getBuffer(BUFFER_IDX) => {Bs}:>BytesResult ]]
-      <bufferHeap> ... BUFFER_IDX |-> Bs:Bytes ... </bufferHeap>
+    rule [[ getBuffer(BUFFER_IDX) => {HEAP[BUFFER_IDX]}:>BytesResult ]]
+        <bufferHeap> HEAP </bufferHeap>
+      requires #validBufferId(BUFFER_IDX, HEAP)
+      [preserves-definedness]
```

Comment:
This rule does not preserve definedness (which also means it's invalid) because `#validBufferId` does not check that `HEAP[BUFFER_IDX]` is of type `BytesResult`. The same for the rule below.

Reply:
Good point, thanks. 
Maybe I just revert the changes to these two rules, the list indexing change is the important one.